pkg/types/baremetal/validation/platform_test.go

```diff
@@ -268,11 +269,40 @@ func TestValidatePlatform(t *testing.T) {
 				Hosts(host1().BootMACAddress("").build()).build(),
 			expected: "baremetal.hosts\\[0\\].BootMACAddress: Required value: missing BootMACAddress",
 		},
+		{
+			name: "toofew_hosts",
+			config: installConfig().
+				BareMetalPlatform(
+					platform().Hosts(
+						host1().build()).build()).
+				ControlPlane(
+					machinePool().Replicas(3).build()).
+				Compute(
+					*machinePool().Replicas(2).build(),
+					*machinePool().Replicas(3).build()).build(),
```

Comment:
With really nested values, the excessive calls to `build` are distracting. Is it avoidable? Maybe by using a builder interface, you could have Compute's build call it on the passed arguments, for example?

Reply:
Sounds a good idea, the syntax looks less heavy, pushed it (version with no interface) (could be also extended directly to the test struct).